layers.go

```diff
@@ -228,10 +267,11 @@ func (r *layerStore) Load() error {
 				}
 				names[name] = layers[n]
 			}
-			if pslice, ok := parents[layer.Parent]; ok {
-				parents[layer.Parent] = append(pslice, layers[n])
-			} else {
-				parents[layer.Parent] = []*Layer{layers[n]}
+			if layer.CompressedDigest != "" {
+				compressedsums[layer.CompressedDigest] = append(compressedsums[layer.CompressedDigest], layer.ID)
+			}
+			if layer.UncompressedDigest != "" {
+				uncompressedsums[layer.UncompressedDigest] = append(uncompressedsums[layer.UncompressedDigest], layer.ID)
```

Comment:
If the underlying driver stores the layer purely as a diff, there can conceivably be multiple copies of the same diff in the driver.  I think we can fix that by changing how the image library creates layers when it's committing an image, but I'd want to add supporting logic to the storage library in a different PR.